dev-tools/generate_documentation.sh

```diff
@@ -31,6 +31,9 @@ fi
 
 cd $(dirname "${BASH_SOURCE}")/..
 
+# remove stale doc files
+rm -rf ./docs ./sphinx/ref ./sphinx/ref-ext
```

Comment:
```suggestion
# remove stale doc files
if [ "$1" == "--clean" ]; then
    echo "Removing temporary documentation files."
    rm -rf ./docs ./sphinx/ref ./sphinx/ref-ext
fi
```
This makes the doc generation process significantly slower, so I would only perform this step if absolutely necessary...
Maybe it would also be good to add information about the new `--clean` flag to the documentation itself? But I can also do this in a separate PR.

Reply:
That's ok for me, this command just needed execution for a single time on the remote repo to fix a doc generation exception